src/MSBuild.Conversion.Project/Converter.cs

```diff
@@ -55,6 +54,33 @@ internal IProjectRootElement ConvertProjectFile(string? defaultTFM, bool keepCur
                 .AddItemRemovesForIntroducedItems(_differs)
                 .RemoveUnnecessaryTargetsIfTheyExist()
                 .ModifyProjectElement();
+
+            static string GetBestTFM(BaselineProject baselineProject, bool keepCurrentTfm, string? specifiedTFM, bool usePreviewSDK)
+            {
+                if (string.IsNullOrWhiteSpace(specifiedTFM))
+                {
+                    // Let's figure this out, friends
+                    var tfmForApps = TargetFrameworkHelper.FindHighestInstalledTargetFramework(usePreviewSDK);
+
+                    specifiedTFM =
+                        keepCurrentTfm
+                        ? baselineProject.GetTfm()
```

Comment:
![image](https://media.giphy.com/media/xiMUwBRn5RDLhzwO80/giphy.gif)

Reply:
Can we use some `if` statements? ternmary straigh into a double nested switch for one of the more complex code paths is going to be hard to debug

---
In reply to: [422376966](https://github.com/dotnet/try-convert/pull/291#discussion_r422376966) [](ancestors = 422376966)